crates/goose/src/session/storage.rs

```diff
@@ -1294,52 +1294,15 @@ pub async fn generate_description_with_schedule_id(
         ));
     }
 
-    // Create a special message asking for a 3-word description
-    let mut description_prompt = "Based on the conversation so far, provide a concise description of this session in 4 words or less. This will be used for finding the session later in a UI with limited space - reply *ONLY* with the description".to_string();
-
-    // get context from messages so far, limiting each message to 300 chars for security
-    let context: Vec<String> = messages
-        .iter()
-        .filter(|m| m.role == rmcp::model::Role::User)
-        .take(3) // Use up to first 3 user messages for context
-        .map(|m| {
-            let text = m.as_concat_text();
-            safe_truncate(&text, 300)
-        })
-        .collect();
-
-    if !context.is_empty() {
-        description_prompt = format!(
-            "Here are the first few user messages:\n{}\n\n{}",
-            context.join("\n"),
-            description_prompt
-        );
-    }
-
-    // Generate the description with error handling
-    let message = Message::user().with_text(&description_prompt);
-    let result = provider
-        .complete(
-            "Reply with only a description in four words or less",
-            &[message],
-            &[],
-        )
+    // Use the provider's session naming capability
+    let sanitized_description = provider
+        .generate_session_name(messages)
```

Comment:
much better! still not really where it should be called from, but we can clean that up some other time